### DISCUSSION TITLE
Plans for Linux arm64 bundles?

### DISCUSSION THREAD

**Discussion by yveszoundi** [fltk-rs/fltk-rs#1282] [Q&A] [answered]
After the recent addition of the arm64 Mac OS artifact, are you also planning on providing a Linux arm64 archive?

My current local workflow is as follow for Linux arm64 apps (builds from an amd64 Ubuntu VM, iMac host)
1. Ensure that arm64 images can be run with podman
`apt install -y binfmt-support qemu-user-static`

2. Temporarily update the cargo manifest of my project 
`sed` or equivalent Perl script to change `fltk-bundled` with `fltk-shared` in the `Cargo.toml` file

3. Prepare the arm64 build environment
Start a container image and then install typical dependencies (`cmake`, `libx11-dev`, `libxcb related`, etc.)
`podman run -it --rm --volume "${PWD}":/src  docker.io/arm64v8/rust:1.63.0-buster /bin/bash` 

4. Build applications leveraging fltk-rs
The rust flags below are not required to build `fltk-rs` itself.
`RUSTFLAGS='-C link-arg=-lgcc -C link-arg=-lX11' cargo build --target aarch64-unknown-linux-gnu --release`

6. Additional involved work to build an arm64 `AppImage` inside podman with `linuxdeploy`.

**Comment by MoAlyousef** [comment, marked as answer]
Hi

Yes. Ideally it should be buildable using github actions, this allows me to avoid having to build locally. 
Unfortunately github actions itself isn't capable of adding other arches, so this is relegated to docker via gha. I'm testing things out currently here:
https://github.com/MoAlyousef/test_arm

**Reply by MoAlyousef** [reply]
I have added a linux arm64 bundle which would be chosen when using the fltk-bundled feature when building for aarch64-unknown-linux-gnu. (On ubuntu) It requires installing `gcc-aarch64-linux-gnu` for the linker, and setting a `.cargo/config.toml` with:
```toml
[target.aarch64-unknown-linux-gnu]
linker = "aarch64-linux-gnu-gcc"
```

**Reply by yveszoundi** [reply]
Thanks, I'll experiment with it soon.

I will also double-check if it's possible to use environment variables instead of a `config.toml` (i.e. `CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER`). While it's not a big deal, environment variables feel more convenient.
